File: BubbleSort.js
function BubbleSort(arr){
	var sorted = false;
	while (!sorted){
		sorted = true;
		for(i = arr.length-1; i > 0; i--){
			if (arr[i] < arr[i-1]){
				var temp = arr[i];
				arr[i] = arr[i - 1];
				arr[i - 1] = temp;
				sorted = false;
			}
		}
	}
}


var arr = [7,1,3,4,8,6,2,9,7,7,5,1,2,10,12,5,6,2,10,9];

console.time("runTime");
BubbleSort(arr);
console.timeEnd("runTime");
console.log(arr);

File: Python/sorts/bubbleSort.py
import random

def bubbleSort(arr):
	for i in range(len(arr)):
		sorted = True
		for j in range(len(arr)-1-i):
			if arr[j] > arr[j+1]:
				arr[j], arr[j+1] = arr[j+1], arr[j]
				sorted = False
		if sorted: break
	return arr

# arr = [random.randint(0,1000) for _ in range(1000)]
# print bubbleSort([])
